### ISSUE TITLE
Firebird 3.x: How IDENTITY field should be retrieved on an INSERT?

### ISSUE THREAD

**Issue by pomgui** [hgourvest/node-firebird#215]
Almost all the tables in our database have their primary keys defined as IDENTITY, but the insert returns nothing (undefined). It should be necessary to return something like [mysql](https://www.npmjs.com/package/mysql#getting-the-id-of-an-inserted-row) does.

### Example

1. Create a new table:

```sql
CREATE TABLE tmp(
	id int GENERATED BY DEFAULT AS IDENTITY NOT NULL PRIMARY KEY, 
	dummy varchar(10)
);
```
2. Insert a new record

```javascript
const FB = require('node-firebird')
FB.attach(options, (err, db)=>{
  db.query(`INSERT INTO tmp(dummy) VALUES('x')`, [], (err, rows)=>{
    console.log('Insert returns: ', rows);
    db.query(`SELECT * FROM tmp`, [], (err, rows)=>{
      console.log('Select returns: ', rows);
    });
  });
  db.detach(err=>{});
});
```

This will print:
```
Insert returns:  undefined
Select returns:  [ { id: 1, dummy: 'x' } ]
```
So, how can we retrieve the new record's PK? Perhaps we're missing something?

**Comment by maratth** [comment]
You can return value from insert like `INSERT INTO table(field) VALUES ('x') RETURNING id`

**Comment by pomgui** [comment]
Thank you @maratth! 
After I wrote this issue I finally found that info in this [FAQ](http://www.firebirdfaq.org/faq243/) page, and immediately I tested it with the same code (only adding "returning id" just as you suggested), but this was the result:
```
Insert returns:  { id: 7 }
Thrown:
TypeError: Cannot read property 'length' of undefined
    at describe (/home/wpomier/progs/pi-database/node_modules/node-firebird/lib/index.js:3768:31)
    at mainCallback (/home/wpomier/progs/pi-database/node_modules/node-firebird/lib/index.js:3705:4)
    at doCallback (/home/wpomier/progs/pi-database/node_modules/node-firebird/lib/index.js:1299:5)
```
As it already was 4AM in my TZ, I didn't edit this issue and I went to sleep :-)

Can you see something wrong with my code?

**Comment by maratth** [comment]
Yes you need to detach database in your query callback not in attach callback.

**Comment by pomgui** [comment]
@maratth, @mariuz, Sorry, somehow I've missed your answer, but @maratth is wrong.
In the same connection (with an explicit or implicit transaction) you're supposed being able to execute as many queries as you want and only then commit and detach from the database.

I really think this is a bug in the driver, but as this issue was closed, I've opened a new issue #219 with a better title and execution examples.
